### DISCUSSION TITLE
How can I use code-server on HPC ?

### DISCUSSION THREAD

**Discussion by illumination-k** [coder/code-server#3273] [Q&A] [answered]
Hi,

I would like to use code-server on HPC.

I tried using singularity from docker and I got the error `error got error from VS Code {"error":{"errno":"ENOENT","code":"ENOENT","syscall":"connect","address":"/run/user/my_id/vscode-ipc-e8693fd6-92b4-4ff2-94ec-c3d67a892d82.sock"}}`

I thought that there is no file of socket (maybe because of permission ?), but I cannot solve how to generate this sock file or change the directory which sock file is generated.

I found that `createRandomIPCHandle` in [lib/vscode/src/vs/base/parts/ipc/node/ipc.net.ts](https://github.com/cdr/code-server/blob/e9985bd658ac42c5bef38553b3e83fdb1e459be3/lib/vscode/src/vs/base/parts/ipc/node/ipc.net.ts) make the sock file, and I export XDG_RUNTIME_DIR Path, but I got same error message.

I also tried run binary, but I got the same error message.

Thanks,

**Comment by code-asher** [comment, marked as answer]
Interesting, does `/run/user/$UID` not exist?

How are you exporting `XDG_RUNTIME_DIR`? Does something like this work?

```
XDG_RUNTIME_DIR=/tmp code-server
```

**Reply by illumination-k** [reply]
Sorry, I found that I can solve this problem that when using the default mac terminal.

I always connect the login node of HPC by VScode Remote SSH, and in that case, I got the above error. It seems conflict or something? But I wonder why even if I use singularity, these problems occurred. In the VScode Remote SSH environment, `XDG_RUNTIME_DIR=/tmp code-server` cannot fix it both of singularity and binary.

**Reply by code-asher** [reply]
By "Remote SSH" do you mean the remote SSH plugin? (This one here: https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-ssh) If so, that's a completely different thing from code-server.

If not then it might help to describe how you are connecting to and starting code-server on the remote machine.

**Reply by illumination-k** [reply]
Thanks for your response. I know the `Remote SSH plugin` is totally different from `code-server`.

The work I did was as follows,

1. connect "login node"
2. run `code-server` on "computing node" with qsub for computing resources
3. connect `code-server` with ssh, like `ssh -NL localport:node_name:node_port` from local machine.

I usually connect to "login node" by ssh with `Remote SSH plugin`, and I tried to run `code-server` on "login node" and "computing node" with qsub and failed.
In that case, I got the error in both singularity and binary.

When I connected by ssh to "login node" with the default terminal instead of "Remote SSH", I can run `code-server` on "login node" and "computing node" in both singularity and binary.

**Reply by code-asher** [reply]
Ahhhhh OK I think I understand.

I guess `XDG_RUNTIME_DIR` must be getting set in the Remote SSH environment somehow but not in the default terminal environment.

I'm not sure what singularity is, but if it's Docker it does seem confusing that you would still have an error in there. Does singularity forward the environment variables somehow, maybe?

**Reply by illumination-k** [reply]
Maybe you are right. "singularity" forward the environment variables ([link](https://sylabs.io/guides/3.7/user-guide/environment_and_metadata.html)). I run `code-server` on singularity with `-e` flag to reset the environment variables of host, and I succeeded to run `code-server` on Remote SSH environment!
